src/codec.rs

```diff
@@ -125,78 +154,126 @@ fn number_of_digits(mut n: usize) -> usize {
     num_digits
 }
 
-impl<T: DeserializeOwned> Decoder for LanguageServerCodec<T> {
-    type Item = T;
+impl<T: DeserializeOwned + std::fmt::Debug> Decoder for LanguageServerCodec<T> {
     type Error = ParseError;
+    type Item = T;
```

Comment:
Very small nitpick, but perhaps we could keep the same ordering of associated types as the upstream definition of the `Decoder` trait, for consistency?

Reply:
I see that the associated type ordering in the `Decoder` impl hasn't been addressed yet?